### ISSUE TITLE
Next.js 15 requires `params` in Route Handlers to be awaited

### ISSUE THREAD

**Issue by AmruthPillai** [zenstackhq/zenstack#1797]
I'm using the latest `@zenstackhq/server/next` package, which exports the `NextRequestHandler`.

I get this error while running the server: 
```
Error:  In route /api/model/[...path] a param property was accessed directly with `params.path`. `params` should be awaited before accessing its properties. Learn more: https://nextjs.org/docs/messages/sync-dynamic-apis`.
```

Probably NextRequestHandler should be updated to await the `params` props before being accessed. AFAIK, I couldn't find a way to await it myself.

**Comment by ymc9** [comment]
Thanks for bringing this up @AmruthPillai ! We should fix it soon. Would you be interested in making a PR for it? I'll be happy to facilitate.

**Comment by AmruthPillai** [comment]
@ymc9 I have raised PR for the same, followed the contribution guide to match the right branch to fork from (`dev`). Hope it looks alright.

**Comment by ymc9** [comment]
Oh, I think it requires the type of the second parameter to be entirely a promise now. I'll make a change.

**Comment by ymc9** [comment]
Fixed in 2.7.4

**Comment by AmruthPillai** [comment]
@ymc9 Thank you for taking over. I was just wondering what would be the compatibility for users using Zenstack pre-Next.js 15, but as long as it works, that's perfect. Looking forward to seeing this in the release!
